src/types/Writable.ts

```diff
@@ -15,6 +15,12 @@ type WritableKeys<T> = {
 	[P in keyof T]: IfEquals<{ [Q in P]: T[P] }, { -readonly [Q in P]: T[P] }, P>;
 }[keyof T];
 
+type OptionalKeys<T> = {
+	[P in keyof T]: {} extends Pick<T, P> ? P : never;
+}[keyof T];
+
+type RequiredKeys<T> = Exclude<keyof T, OptionalKeys<T>>;
```

Comment:
Likewise, this is a union type of all the remaining keys, which by definition are not optional.